### PR TITLE
Issue #878 permit configuration of io threads

### PR DESCRIPTION
**Pull request by jmjatlanta**
Issue #878 

This PR implements a configuration parameter (config.ini or command line) that allows for control of the number of IO threads.

It works in conjunction with changes in bitshares-fc (https://github.com/bitshares/bitshares-fc/pull/47) that permits changes to the default number of io threads, and also attempts a better guess of how many io threads should be created.

### CODE REVIEW

**Comment by jmjatlanta** [comment]
@pmconrad Yes, I know for a fact that the delayed_plugin starts the default_io_service during registry. That's why I moved it. It seems strange to me that a plugin can change parameters, and plugins don't need a configuration file to get started. But I guess that's the way it is.

Options I can think of:
- Put things back the way they were, and do not allow setting io-threads in the config file (only on the command line). Set the number of IO threads early, call default_io_service as part of main(). The plugins will then use the existing "singleton".
- Do not permit plugins to fire up the default_io_service until later in the startup process.
- Have plugins modify some collection of options that can then be merged into the options from the config file and command line.

I'm sure there are other options, just typing out loud.

**Comment by pmconrad** [comment]
Plugins can't change parameter settings, but they add their own parameters to the list of known parameters - https://github.com/bitshares/bitshares-core/blob/master/libraries/app/include/graphene/app/application.hpp#L69

Plugins *shouldn't* do more than absolutely necessary during registration, and perform required initialization in plugin_initialize. So I'd say move the registration back to where it was, and modify delayed node to create its impl (if that's the culprit) when plugin_initialize is called.

**Comment by jmjatlanta** [comment]
delayed_node seems to have been the only culprit.

File: programs/witness_node/main.cpp [hunk 1]
```diff
@@ -244,6 +245,13 @@ int main(int argc, char** argv) {
          create_new_config_file( config_ini_path, data_dir, cfg_options );
       load_config_file( config_ini_path, cfg_options, options );
 
+      if ( options.count("io-threads") )
```
Comment (pmconrad) [inline programs/witness_node/main.cpp]: Please move this into application::initialize. (That has the added advantage that the option will also be interpreted correctly by the delayed_node executable.)

Comment (jmjatlanta) [reply programs/witness_node/main.cpp]: Done

**Comment by abitmore** [comment]
I guess the "bump fc" commit is wrong. Is it here for making travis work? We always need to use a commit in master branch of fc. We can correct this after the fc PR is merged.

**Comment by pmconrad** [comment]
Tested ok, please bump fc again after merging https://github.com/bitshares/bitshares-fc/pull/47, then I'll approve.

**Comment by jmjatlanta** [comment]
Yes, the bump was to make cloning easier and travis happy. In the future, I'll keep the fc submodule pointing to master.

**Comment by abitmore** [comment]
@jmjatlanta @pmconrad I changed description of the new parameter. Is it OK?

**Pull request merged**